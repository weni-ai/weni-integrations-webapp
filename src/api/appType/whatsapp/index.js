import qs from 'query-string';
import request from '@/api/request.js';

const resource = '/api/v1/apptypes';
const templatesResource = '/api/v1/apps';

export default {
  async fetchWppContactInfo(appCode, appUuid) {
    return await request.$http
      .get(`${resource}/${appCode}/apps/${appUuid}/contact/`)
      .then((r) => r.data);
  },
  updateWppContactInfo(appCode, appUuid, payload) {
    return request.$http.patch(`${resource}/${appCode}/apps/${appUuid}/contact/`, payload);
  },
  getConversations(appCode, appUuid, params) {
    const queryString = qs.stringify(params);
    return request.$http.get(
      `${resource}/${appCode}/apps/${appUuid}/conversations/?${queryString}`,
    );
  },
  fetchWppProfile(appCode, appUuid) {
    return request.$http.get(`${resource}/${appCode}/apps/${appUuid}/profile/`);
  },
  updateWppProfile(appCode, appUuid, data) {
    return request.$http.patch(`${resource}/${appCode}/apps/${appUuid}/profile/`, data);
  },
  deleteWppProfilePhoto(appCode, appUuid) {
    return request.$http.delete(`${resource}/${appCode}/apps/${appUuid}/profile/`);
  },
  async getWhatsAppTemplates(appUuid, params) {
    const queryString = qs.stringify(params);
    return await request.$http
      .get(`${templatesResource}/${appUuid}/templates/?${queryString}`)
      .then((r) => r.data);
  },
  async fetchTemplateData(appUuid, templateUuid) {
    return await request.$http
      .get(`${templatesResource}/${appUuid}/templates/${templateUuid}/`)
      .then((r) => r.data);
  },
  async fetchSelectLanguages(appUuid) {
    return await request.$http
      .get(`${templatesResource}/${appUuid}/templates/languages/`)
      .then((r) => r.data);
  },
  async createTemplate(appUuid, data) {
    return request.$http
      .post(`${templatesResource}/${appUuid}/templates/`, data)
      .then((r) => r.data);
  },
  deleteTemplateMessage(appUuid, templateUuid) {
    return request.$http.delete(`${templatesResource}/${appUuid}/templates/${templateUuid}`);
  },
  async createTemplateTranslation(appUuid, templateUuid, data) {
    return await request.$http
      .post(`${templatesResource}/${appUuid}/templates/${templateUuid}/translations/`, data)
      .then((r) => r.data);
  },
  updateTemplateTranslation(appUuid, templateUuid, data) {
    return request.$http.patch(`${templatesResource}/${appUuid}/templates/${templateUuid}/`, data);
  },
  updateWppWebhookInfo(appCode, appUuid, payload) {
    return request.$http.patch(`${resource}/${appCode}/apps/${appUuid}/update_webhook/`, payload);
  },
  async requestConversationsReport(appCode, appUuid, params) {
    const queryString = qs.stringify(params);
    return await request.$http.get(
      `${resource}/${appCode}/apps/${appUuid}/report_sent_messages/?${queryString}`,
    );
  },
};
