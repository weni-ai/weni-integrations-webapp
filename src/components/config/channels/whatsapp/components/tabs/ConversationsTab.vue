<template>
  <div class="conversations">
    <div class="conversations__content">
      <div :class="['conversations__content__dropdown', { active: showDateFilter }]">
        <div class="conversations__content__label">
          {{ $t('WhatsApp.config.conversations.filter.label') }}
        </div>
        <unnnic-date-filter
          class="conversations__content__dropdown-filter"
          dateFormat="DD/MM/YYYY"
          @filter="showDateFilter = true"
          :startDate="startDateObject"
          :endDate="endDateObject"
          placeholder="DD/MM/YYYY ~ DD/MM/YYYY"
        />

        <div class="conversations__content__dropdown-data">
          <unnnic-date-picker
            v-if="showDateFilter"
            size="small"
            :days="['D', 'S', 'T', 'Q', 'Q', 'S', 'S']"
            @change="handleDateFilter"
          />
        </div>
      </div>

      <div class="conversations__content__tables">
        <div class="conversations__content__label">
          {{ $t('WhatsApp.config.conversations.conversations_count.label') }}
        </div>

        <div class="conversations__content__tables__section">
          <span v-if="hasBeforeAndAfter">
            {{ $t('WhatsApp.config.conversations.conversations_count.before_label') }}
          </span>
          <unnnic-table v-if="hasBefore" :items="beforeItems">
            <template v-slot:header>
              <unnnic-table-row :headers="beforeHeaders" />
            </template>

            <template v-slot:item="{ item }">
              <unnnic-table-row :headers="beforeHeaders">
                <template v-slot:startedBy>
                  <div :title="item.startedBy" class="break-text">
                    {{ item.startedBy }}
                  </div>
                </template>

                <template v-slot:quantity>
                  <div :title="item.quantity" class="break-text">
                    {{ item.quantity }}
                  </div>
                </template>
              </unnnic-table-row>
            </template>
          </unnnic-table>
        </div>

        <div class="conversations__content__tables__section">
          <span v-if="hasBeforeAndAfter">
            {{ $t('WhatsApp.config.conversations.conversations_count.after_label') }}
          </span>
          <unnnic-table v-if="(!hasBefore && !hasAfter) || hasAfter" :items="afterItems">
            <template v-slot:header>
              <unnnic-table-row :headers="afterHeaders" />
            </template>

            <template v-slot:item="{ item }">
              <unnnic-table-row :headers="afterHeaders">
                <template v-slot:category>
                  <div :title="item.category" class="break-text">
                    {{ item.category }}
                  </div>
                </template>

                <template v-slot:quantity>
                  <div :title="item.quantity" class="break-text">
                    {{ item.quantity }}
                  </div>
                </template>
              </unnnic-table-row>
            </template>
          </unnnic-table>
        </div>
      </div>

      <div class="conversations__content__report">
        <span class="conversations__content__report__label">
          {{ $t('WhatsApp.config.conversations.report.label') }}
        </span>

        <span class="conversations__content__report__description">
          {{ $t('WhatsApp.config.conversations.report.description') }}
        </span>

        <unnnic-tool-tip
          class="conversations__content__report__tooltip"
          :text="$t('WhatsApp.config.conversations.report.missing_dates')"
          :enabled="!this.startDate || !this.endDate"
          side="top"
          maxWidth="15rem"
        >
          <unnnic-button
            ref="reportButton"
            class="conversations__content__report__button"
            type="secondary"
            size="small"
            @click="requestReport"
            :disabled="!this.startDate || !this.endDate"
          >
            {{ $t('WhatsApp.config.conversations.report.button') }}
          </unnnic-button>
        </unnnic-tool-tip>
      </div>
    </div>
    <div class="conversations__buttons">
      <unnnic-button
        class="conversations__buttons__close"
        type="tertiary"
        size="large"
        :text="$t('general.Cancel')"
        @click="() => this.$emit('close')"
      />
    </div>
  </div>
</template>

<script>
  import debounce from 'lodash.debounce';
  import unnnic from '@weni/unnnic-system';
  import { mapActions, mapState } from 'pinia';
  import { auth_store } from '@/stores/modules/auth.store';
  import { whatsapp_store } from '@/stores/modules/appType/channels/whatsapp.store';
  import removeEmpty from '../../../../../../utils/clean';

  export default {
    name: 'ConversationsTab',
    props: {
      app: {
        type: Object,
        default: /* istanbul ignore next */ () => {},
      },
    },
    data() {
      return {
        showDateFilter: false,
        startDate: null,
        endDate: null,
        businessInitiated: 0,
        userInitiated: 0,
        totalInitiated: 0,
        afterData: {},
        beforeHeaders: [
          {
            id: 'startedBy',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.started_by'),
            flex: 1,
          },
          {
            id: 'quantity',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.quantity'),
            flex: 1,
          },
        ],
        afterHeaders: [
          {
            id: 'category',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.category'),
            flex: 1,
          },
          {
            id: 'quantity',
            text: this.$t('WhatsApp.config.conversations.conversations_count.table.quantity'),
            flex: 1,
          },
        ],
      };
    },
    created() {
      /* istanbul ignore next */
      window.addEventListener('click', (event) => {
        if (event.target.closest('.conversations__content__dropdown')) {
          return false;
        }
        this.showDateFilter = false;
      });
    },
    computed: {
      ...mapState(auth_store, ['project']),
      ...mapState(whatsapp_store, [
        'whatsAppConversations',
        'loadingConversations',
        'errorConversations',
        'loadingConversationsReport',
        'errorConversationsReport',
      ]),
      startDateObject() {
        return this.startDate && new Date(this.startDate.replace('-', ' '));
      },
      endDateObject() {
        return this.endDate && new Date(this.endDate.replace('-', ' '));
      },
      beforeItems() {
        return [
          {
            startedBy: this.$t('WhatsApp.config.conversations.conversations_count.table.business'),
            quantity: this.businessInitiated || 0,
          },
          {
            startedBy: this.$t('WhatsApp.config.conversations.conversations_count.table.user'),
            quantity: this.userInitiated || 0,
          },
        ];
      },
      afterItems() {
        return [
          {
            category: this.$t('WhatsApp.config.conversations.conversations_count.table.marketing'),
            quantity: this.afterData?.MARKETING || 0,
          },
          {
            category: this.$t('WhatsApp.config.conversations.conversations_count.table.marketing_lite'),
            quantity: this.afterData?.MARKETING_LITE || 0,
          },
          {
            category: this.$t('WhatsApp.config.conversations.conversations_count.table.utility'),
            quantity: this.afterData?.UTILITY || 0,
          },
          {
            category: this.$t(
              'WhatsApp.config.conversations.conversations_count.table.authentication',
            ),
            quantity: this.afterData?.AUTHENTICATION || 0,
          },
          {
            category: this.$t('WhatsApp.config.conversations.conversations_count.table.service'),
            quantity: this.afterData?.SERVICE || 0,
          },
        ];
      },
      hasBeforeAndAfter() {
        return this.hasBefore && this.hasAfter;
      },
      hasBefore() {
        return !!this.beforeItems.find((item) => item.quantity > 0);
      },
      hasAfter() {
        return !!this.afterItems.find((item) => item.quantity > 0);
      },
    },
    methods: {
      ...mapActions(whatsapp_store, ['getConversations', 'requestConversationsReport']),
      handleDateFilter: debounce(async function (event) {
        this.startDate = event.startDate;
        this.endDate = event.endDate;

        const params = {
          start: event.startDate,
          end: event.endDate,
        };

        await this.getConversations({
          code: this.app.code,
          appUuid: this.app.uuid,
          params,
        });

        if (this.errorConversations) {
          const err =
            this.errorConversations?.error_user_msg ||
            this.$t('WhatsApp.config.conversations.fetch_error');
          unnnic.unnnicCallAlert({
            props: {
              text: err,
              type: 'error',
            },
            seconds: 6,
          });
          return;
        }

        this.businessInitiated = this.whatsAppConversations.business_initiated;
        this.userInitiated = this.whatsAppConversations.user_initiated;
        this.totalInitiated = this.whatsAppConversations.total;
        this.afterData = this.whatsAppConversations.templates;
      }, 750),
      async requestReport() {
        const params = removeEmpty({
          project_uuid: this.project,
          start_date: this.startDate,
          end_date: this.endDate,
        });

        await this.requestConversationsReport({
          code: this.app.code,
          appUuid: this.app.uuid,
          params,
        });

        if (this.errorConversationsReport) {
          let errorText =
            this.errorConversationsReport?.error_user_msg ||
            this.$t('WhatsApp.config.conversations.report_error');

          if (this.errorConversationsReport.response?.status === 409) {
            errorText = this.$t('WhatsApp.config.conversations.report_already_processing');
          }
          unnnic.unnnicCallAlert({
            props: {
              text: errorText,
              type: 'error',
            },
            seconds: 6,
          });
          return;
        }

        unnnic.unnnicCallAlert({
          props: {
            text: this.$t('WhatsApp.config.conversations.report_success'),
            type: 'success',
          },
          seconds: 6,
        });
        return;
      },
    },
  };
</script>

<style lang="scss" scoped>
  .conversations {
    display: flex;
    flex-direction: column;
    flex: 1;
    margin-top: $unnnic-spacing-inline-xs;

    &__content {
      display: flex;
      flex-direction: column;
      flex: 1;
      gap: $unnnic-spacing-stack-md;

      &__dropdown {
        width: fit-content;
        &-data {
          position: absolute;
          z-index: 1;
        }

        &-filter {
          width: 14rem;
        }
      }

      &__label {
        font-weight: $unnnic-font-weight-bold;
        font-size: $unnnic-font-size-body-gt;
        line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;
        color: $unnnic-color-neutral-darkest;
      }

      &__report {
        display: flex;
        flex-direction: column;
        gap: $unnnic-spacing-stack-sm;
        font-size: $unnnic-font-size-body-gt;
        line-height: $unnnic-font-size-body-gt + $unnnic-line-height-md;

        &__label {
          font-weight: $unnnic-font-weight-bold;
          color: $unnnic-color-neutral-darkest;
        }

        &__description {
          font-weight: $unnnic-font-weight-regular;
          color: $unnnic-color-neutral-cloudy;
        }

        &__tooltip {
          width: 33%;
        }

        &__button {
          width: 100%;
        }
      }

      &__tables {
        display: flex;
        flex-direction: column;
        gap: $unnnic-spacing-stack-sm;

        &__section {
          display: flex;
          flex-direction: column;
          gap: $unnnic-spacing-stack-xs;
          ::v-deep .scroll {
            padding-right: 0;
          }
        }
      }
    }

    &__buttons {
      display: flex;
      justify-content: flex-end;
      margin-top: $unnnic-spacing-stack-sm;

      &__close {
        width: 50%;
      }
    }
  }
</style>
