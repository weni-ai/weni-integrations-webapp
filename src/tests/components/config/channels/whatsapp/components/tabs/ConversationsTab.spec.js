import { mount } from '@vue/test-utils';
import ConversationsTab from '@/components/config/channels/whatsapp/components/tabs/ConversationsTab.vue';
import { createTestingPinia } from '@pinia/testing';
import { setActivePinia } from 'pinia';
import { whatsapp_store } from '@/stores/modules/appType/channels/whatsapp.store';
import unnnic from '@weni/unnnic-system';
import i18n from '@/utils/plugins/i18n';
import UnnnicSystem from '@/utils/plugins/UnnnicSystem';
import { describe, it, beforeEach, expect, vi } from 'vitest';

describe('ConversationsTab', () => {
  let wrapper;
  const pinia = createTestingPinia({ stubActions: false });
  setActivePinia(pinia);

  beforeEach(() => {
    wrapper = mount(ConversationsTab, {
      props: {
        app: { code: 'app_code', uuid: 'app_uuid' },
      },
      global: {
        plugins: [i18n, UnnnicSystem, pinia],
        mocks: {
          $t: (key) => key,
          $router: { replace: vi.fn() },
          unnnic,
        },
      },
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the filter label correctly', () => {
    expect(wrapper.find('.conversations__content__label').text()).toBe('Filter by');
  });

  it('displays the conversations count labels correctly', () => {
    const labels = wrapper.findAll('.conversations__content__label');
    expect(labels[0].text()).toBe('Filter by');
    expect(labels[1].text()).toBe('Number of conversations');
  });

  it('calls requestReport when button is clicked', async () => {
    const requestReportSpy = vi.spyOn(wrapper.vm, 'requestReport');
    const button = wrapper.findComponent({ ref: 'reportButton' });
    expect(button.exists()).toBe(true);
    wrapper.vm.startDate = '2024-01-01';
    wrapper.vm.endDate = '2024-01-04';
    await wrapper.vm.$nextTick();

    expect(button.props().disabled).toBe(false);
    await button.trigger('click');
    expect(requestReportSpy).toHaveBeenCalled();
  });

  it('shows alert on error during reporting', async () => {
    const spyCallAlert = vi.spyOn(unnnic, 'unnnicCallAlert');
    const store = whatsapp_store();
    store.errorConversationsReport = {
      error_user_msg: 'Report Error',
      response: { status: 500 },
    };
    await wrapper.vm.requestReport();

    expect(spyCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'Failed to request detailed report, please try again later',
        type: 'error',
      },
      seconds: 6,
    });
  });

  it('shows alert for report already processing', async () => {
    const spyCallAlert = vi.spyOn(unnnic, 'unnnicCallAlert');
    const store = whatsapp_store();
    store.errorConversationsReport = { response: { status: 409 } };
    await wrapper.vm.requestReport();

    expect(spyCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'Failed to request detailed report, please try again later',
        type: 'error',
      },
      seconds: 6,
    });
  });

  it('shows success alert on successful report', async () => {
    const spyCallAlert = vi.spyOn(unnnic, 'unnnicCallAlert');
    const store = whatsapp_store();
    store.requestConversationsReport = vi.fn().mockResolvedValue();
    store.errorConversationsReport = null;
    expect(wrapper.vm.errorConversationsReport).toBe(null);

    await wrapper.vm.requestReport();

    expect(spyCallAlert).toHaveBeenCalledWith({
      props: {
        text: 'You will receive your detailed report by email shortly',
        type: 'success',
      },
      seconds: 6,
    });
  });

  it('computes beforeItems correctly', () => {
    wrapper.vm.businessInitiated = 5;
    wrapper.vm.userInitiated = 10;

    expect(wrapper.vm.beforeItems).toEqual([
      { startedBy: 'Business', quantity: 5 },
      { startedBy: 'User', quantity: 10 },
    ]);
  });

  it('computes afterItems correctly', () => {
    wrapper.vm.afterData = {
      MARKETING: 3,
      MARKETING_LITE: 5,
      UTILITY: 4,
      AUTHENTICATION: 2,
      SERVICE: 1,
    };

    expect(wrapper.vm.afterItems).toEqual([
      { category: 'Marketing', quantity: 3 },
      { category: 'Marketing Lite', quantity: 5 },
      { category: 'Utility', quantity: 4 },
      { category: 'Authentication', quantity: 2 },
      { category: 'Service', quantity: 1 },
    ]);
  });

  it('computes hasBefore and hasAfter correctly', () => {
    wrapper.vm.businessInitiated = 0;
    wrapper.vm.userInitiated = 0;
    expect(wrapper.vm.hasBefore).toBe(false);

    wrapper.vm.afterData = {
      MARKETING: 3,
      UTILITY: 0,
      AUTHENTICATION: 0,
      SERVICE: 0,
    };
    expect(wrapper.vm.hasAfter).toBe(true);
  });

  it('computes startDateObject and endDateObject correctly', () => {
    wrapper.vm.startDate = '2024-01-01';
    wrapper.vm.endDate = '2024-01-02';

    const startDate = wrapper.vm.startDateObject;
    const endDate = wrapper.vm.endDateObject;

    expect(startDate.getUTCFullYear()).toEqual(2024);
    expect(startDate.getUTCMonth()).toEqual(0);
    expect(startDate.getUTCDate()).toEqual(1);

    expect(endDate.getUTCFullYear()).toEqual(2024);
    expect(endDate.getUTCMonth()).toEqual(0);
    expect(endDate.getUTCDate()).toEqual(2);
  });
});
