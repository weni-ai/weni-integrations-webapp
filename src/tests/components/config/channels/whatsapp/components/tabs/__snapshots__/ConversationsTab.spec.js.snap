// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConversationsTab > matches the snapshot 1`] = `
"<div data-v-2b8fe4c5="" class="conversations">
  <div data-v-2b8fe4c5="" class="conversations__content">
    <div data-v-2b8fe4c5="" class="conversations__content__dropdown">
      <div data-v-2b8fe4c5="" class="conversations__content__label">Filter by</div>
      <div data-v-2b8fe4c5="" class="unnnic-date-filter conversations__content__dropdown-filter">
        <div data-v-fefd47bc="" class="unnnic-form sm unnnic-date-filter__input" readonly="" value="">
          <!---->
          <div data-v-c1401a33="" data-v-fefd47bc="" class="text-input size--sm text-input--icon-right-size-sm unnnic-date-filter__input unnnic-form-input" readonly="" mask="##/##/#### !~ ##/##/####" value=""><input data-v-0caa2ccb="" data-v-c1401a33="" class="unnnic-date-filter__input unnnic-form-input input-itself input size-sm normal input--has-icon-left unnnic-date-filter__input unnnic-form-input input-itself" readonly="" placeholder="DD/MM/YYYY ~ DD/MM/YYYY" iconleft="notes-1" iconrightsize="sm" hascloudycolor="false" type="text" value=""><svg data-v-9846e6db="" data-v-c1401a33="" width="40" height="40" viewBox="0 0 40 40" fill="none" xmlns="http://www.w3.org/2000/svg" class="unnnic-icon unnnic-icon__size--sm unnnic-icon-scheme--neutral-cloudy icon-left" data-testid="old-map-icons">
              <path d="M8.75 35.0005C7.71625 35.0005 6.875 34.1592 6.875 33.1255V9.37549C6.875 8.34174 7.71625 7.50049 8.75 7.50049H11.875V5.62549C11.875 5.28049 12.155 5.00049 12.5 5.00049C12.845 5.00049 13.125 5.28049 13.125 5.62549V7.50049H16.875V5.62549C16.875 5.28049 17.155 5.00049 17.5 5.00049C17.845 5.00049 18.125 5.28049 18.125 5.62549V7.50049H21.875V5.62549C21.875 5.28049 22.155 5.00049 22.5 5.00049C22.845 5.00049 23.125 5.28049 23.125 5.62549V7.50049H26.875V5.62549C26.875 5.28049 27.155 5.00049 27.5 5.00049C27.845 5.00049 28.125 5.28049 28.125 5.62549V7.50049H31.25C32.2838 7.50049 33.125 8.34174 33.125 9.37549V33.1255C33.125 34.1592 32.2838 35.0005 31.25 35.0005H8.75ZM8.75 8.75049C8.405 8.75049 8.125 9.03049 8.125 9.37549V33.1255C8.125 33.4705 8.405 33.7505 8.75 33.7505H31.25C31.595 33.7505 31.875 33.4705 31.875 33.1255V9.37549C31.875 9.03049 31.595 8.75049 31.25 8.75049H28.125V10.6255C28.125 10.9705 27.845 11.2505 27.5 11.2505C27.155 11.2505 26.875 10.9705 26.875 10.6255V8.75049H23.125V10.6255C23.125 10.9705 22.845 11.2505 22.5 11.2505C22.155 11.2505 21.875 10.9705 21.875 10.6255V8.75049H18.125V10.6255C18.125 10.9705 17.845 11.2505 17.5 11.2505C17.155 11.2505 16.875 10.9705 16.875 10.6255V8.75049H13.125V10.6255C13.125 10.9705 12.845 11.2505 12.5 11.2505C12.155 11.2505 11.875 10.9705 11.875 10.6255V8.75049H8.75Z" fill="#3B414D" class="primary"></path>
              <path d="M12.5 17.5005C12.155 17.5005 11.875 17.2205 11.875 16.8755C11.875 16.5305 12.155 16.2505 12.5 16.2505H27.5C27.845 16.2505 28.125 16.5305 28.125 16.8755C28.125 17.2205 27.845 17.5005 27.5 17.5005H12.5Z" fill="#3B414D" class="primary"></path>
              <path d="M12.5 21.2505C12.155 21.2505 11.875 20.9705 11.875 20.6255C11.875 20.2805 12.155 20.0005 12.5 20.0005H27.5C27.845 20.0005 28.125 20.2805 28.125 20.6255C28.125 20.9705 27.845 21.2505 27.5 21.2505H12.5Z" fill="#3B414D" class="primary"></path>
              <path d="M12.5 25.0005C12.155 25.0005 11.875 24.7205 11.875 24.3755C11.875 24.0305 12.155 23.7505 12.5 23.7505H27.5C27.845 23.7505 28.125 24.0305 28.125 24.3755C28.125 24.7205 27.845 25.0005 27.5 25.0005H12.5Z" fill="#3B414D" class="primary"></path>
              <path d="M12.5 28.7505C12.155 28.7505 11.875 28.4705 11.875 28.1255C11.875 27.7805 12.155 27.5005 12.5 27.5005H21.25C21.595 27.5005 21.875 27.7805 21.875 28.1255C21.875 28.4705 21.595 28.7505 21.25 28.7505H12.5Z" fill="#3B414D" class="primary"></path>
            </svg>
            <!---->
          </div>
          <!---->
        </div>
      </div>
      <div data-v-2b8fe4c5="" class="conversations__content__dropdown-data">
        <!--v-if-->
      </div>
    </div>
    <div data-v-2b8fe4c5="" class="conversations__content__tables">
      <div data-v-2b8fe4c5="" class="conversations__content__label">Number of conversations</div>
      <div data-v-2b8fe4c5="" class="conversations__content__tables__section">
        <!--v-if-->
        <!--v-if-->
      </div>
      <div data-v-2b8fe4c5="" class="conversations__content__tables__section">
        <!--v-if-->
        <div data-v-b2229274="" data-v-2b8fe4c5="" class="unnnic-table">
          <div data-v-b2229274="" class="header">
            <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
              <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                <div data-v-cb0c6b93="" class="break-text">Category</div>
              </div>
              <div data-v-cb0c6b93="" class="divider"></div>
              <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                <div data-v-cb0c6b93="" class="break-text">Quantity</div>
              </div>
              <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
            </div>
          </div>
          <div data-v-b2229274="">
            <div data-v-b2229274="" class="item-spacer"></div>
          </div>
          <div data-v-b2229274="" class="scroll">
            <div data-v-b2229274="" class="item">
              <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="Marketing" class="break-text">Marketing</div>
                </div>
                <div data-v-cb0c6b93="" class="divider"></div>
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="0" class="break-text">0</div>
                </div>
                <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
              </div>
            </div>
            <div data-v-b2229274="" class="item">
              <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="Marketing Lite" class="break-text">Marketing Lite</div>
                </div>
                <div data-v-cb0c6b93="" class="divider"></div>
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="0" class="break-text">0</div>
                </div>
                <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
              </div>
            </div>
            <div data-v-b2229274="" class="item">
              <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="Utility" class="break-text">Utility</div>
                </div>
                <div data-v-cb0c6b93="" class="divider"></div>
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="0" class="break-text">0</div>
                </div>
                <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
              </div>
            </div>
            <div data-v-b2229274="" class="item">
              <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="Authentication" class="break-text">Authentication</div>
                </div>
                <div data-v-cb0c6b93="" class="divider"></div>
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="0" class="break-text">0</div>
                </div>
                <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
              </div>
            </div>
            <div data-v-b2229274="" class="item">
              <div data-v-cb0c6b93="" data-v-2b8fe4c5="" class="table-row">
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="Service" class="break-text">Service</div>
                </div>
                <div data-v-cb0c6b93="" class="divider"></div>
                <div data-v-cb0c6b93="" class="col" style="flex: 1 1 0%; min-width: 0;">
                  <div data-v-2b8fe4c5="" title="0" class="break-text">0</div>
                </div>
                <div data-v-cb0c6b93="" class="divider" style="display: none;"></div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div data-v-2b8fe4c5="" class="conversations__content__report"><span data-v-2b8fe4c5="" class="conversations__content__report__label">Detailed conversation report</span><span data-v-2b8fe4c5="" class="conversations__content__report__description">You can receive message breakdowns by email of no more than 6 months ago. It is possible that some information in the report is inaccurate. If in doubt, contact your account manager.</span>
      <div data-v-8c61d974="" data-v-2b8fe4c5="" class="unnnic-tooltip conversations__content__report__tooltip"><button data-v-d127031e="" data-v-2b8fe4c5="" class="conversations__content__report__button unnnic-button unnnic-button--size-small unnnic-button--secondary conversations__content__report__button" disabled="">
          <!---->
          <!---->
          <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label">Request report </span>
          <!---->
        </button><span data-v-8c61d974="" class="unnnic-tooltip-label unnnic-tooltip-label-top" style="max-width: 15rem; left: 0px; top: -8px;" data-testid="tooltip-label">A date range is required to be able to request a report <br data-v-8c61d974=""><!----></span></div>
    </div>
  </div>
  <div data-v-2b8fe4c5="" class="conversations__buttons"><button data-v-d127031e="" data-v-2b8fe4c5="" class="conversations__buttons__close unnnic-button unnnic-button--size-large unnnic-button--tertiary conversations__buttons__close">
      <!---->
      <!---->
      <!----><span data-v-d127031e="" class="unnnic-button__label" data-testid="button-label"> Cancel</span>
      <!---->
    </button></div>
</div>"
`;
